### DISCUSSION TITLE
GitHub strategy matrix error: Error reading JToken from JsonReader

### DISCUSSION THREAD

**Discussion by nikhiilll** [community/community#26633] [Actions]
<p>Hello,</p>
<p>I’m trying to pass a matrix of Issue node ids from one job to another, to add all the issues to a project board. The error is getting generated in the line which uses fromJSON in the second job - “add_issues_to_repo” which receives a matrix from the first job - “get_repo_issues”. I’m getting the following error after running the GitHub Workflow:</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/3X/0/b/0b4afe712f779d22af043409a1f9e098fa95ab05.png" data-download-href="/uploads/short-url/1BTVp8UNsIGscHRaAr7p8EZbdEF.png?dl=1" title="image" rel="noopener nofollow ugc"><img src="https://user-images.githubusercontent.com/29837446/181098954-aef60409-d79b-441c-877a-978328e0ac2b.png" alt="image" data-base62-sha1="1BTVp8UNsIGscHRaAr7p8EZbdEF" width="689" height="61" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b4afe712f779d22af043409a1f9e098fa95ab05_2_689x61.png, https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b4afe712f779d22af043409a1f9e098fa95ab05_2_1033x91.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b4afe712f779d22af043409a1f9e098fa95ab05_2_1378x122.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/0/b/0b4afe712f779d22af043409a1f9e098fa95ab05_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use href="#far-image"></use></svg><span class="filename">image</span><span class="informations">2551×226 50.8 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use href="#discourse-expand"></use></svg>
</div></a></div>
<p>Following is the code for the workflow:</p>
<pre><code class="lang-auto">name: Add existing issues to the project board

on: [workflow_dispatch]

jobs:

  get_repo_issues:
    runs-on: ubuntu-latest
    outputs:
      matrix: ${{ steps.set_matrix.outputs.matrix }}
    steps:
      - name: Get repository issues
        id: set-matrix
        env:
          GITHUB_TOKEN: ${{ secrets.GH_TOKEN }}
        run: | 
          echo "::set-output name=matrix::{\"include\":[{\"node_id\":\"I_kwDOHW-0fs5KDASF\"}, {\"node_id\":\"I_kwDOHW-0fs5KC_0B\"}]}"
          echo "${{ steps.set_matrix.outputs.matrix }}"
          
  add_issues_to_repo:
    needs: get_repo_issues
    runs-on: ubuntu-latest
    strategy:
      matrix:
        issue: ${{fromJson(needs.get_repo_issues.outputs.matrix)}}
    steps:
   
#Truncated the code below as the error is getting generated in the line using fromJSON function.
</code></pre>
<p>I followed the actions docs here: <a href="https://docs.github.com/en/actions/learn-github-actions/expressions#fromjson" class="inline-onebox" rel="noopener nofollow ugc">Expressions - GitHub Docs</a></p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="nikhiilll" data-post="1" data-topic="252787">
<div class="title">
<div class="quote-controls"></div>
 nikhiilll:</div>
<blockquote>
<pre><code class="lang-yaml">    strategy:
      matrix:
        issue: ${{fromJson(needs.get_repo_issues.outputs.matrix)}}
</code></pre>
</blockquote>
</aside>
<p>You’re assigning to a matrix variable here, so the data you pass in must be a list of values the <code>issue</code> variable should have in the different matrix iterations. The JSON you create seems to set an issue field, so I guess you want to assign it directly to the <code>matrix</code> property.</p>

**Comment by timothyclarke** [comment]
It seems that if you create the matrix using a HereDoc style syntax it works
eg
```bash
- name: Generate Matrix
   id: generate-matrix
   run: |
     echo "For the matrix step later we need to create the output via a here doc method"
     echo "matrix<<DELIMITER" >> ${GITHUB_OUTPUT}
     echo "{\"include\":[{\"key1\":\"value1\"}]}" >> ${GITHUB_OUTPUT}
     echo "DELIMITER" >> ${GITHUB_OUTPUT}
```
You can then use `fromJson`. Note `echo ${{ steps.generate-matrix.outputs.matrix }} | jq -r .` doesn't seem to work, but your later matrix step will. You can expand the above replacing `DELIMITER` with a random generated string and `cat myfile.json >> ${GITHUB_OUTPUT}` if you want to create the json in some other way
